pkg/imageverifycache/client.go

```diff
@@ -0,0 +1,116 @@
+package imageverifycache
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+)
+
+type cache struct {
+	logger         logr.Logger
+	isCacheEnabled bool
+	maxSize        int64
+	TTL            int64
+}
+
+type Option = func(*cache) error
+
+func New(options ...Option) (Client, error) {
+	cache := &cache{}
+	for _, opt := range options {
+		if err := opt(cache); err != nil {
+			return nil, err
+		}
+	}
+
+	return cache, nil
+}
+
+func DisabledImageVerifyCache() Client {
+	return &cache{
+		logger:         logr.Discard(),
+		isCacheEnabled: false,
+		maxSize:        0,
+		TTL:            0,
+	}
+}
+
+func WithLogger(l logr.Logger) Option {
+	return func(c *cache) error {
+		c.logger = l
+		return nil
+	}
+}
+
+func WithCacheEnableFlag(b bool) Option {
+	return func(c *cache) error {
+		c.isCacheEnabled = b
+		return nil
+	}
+}
+
+func WithMaxSize(s int64) Option {
+	return func(c *cache) error {
+		c.maxSize = s
+		return nil
+	}
+}
+
+func WithTTLDuration(t int64) Option {
+	return func(c *cache) error {
+		c.TTL = t
+		return nil
+	}
+}
+
+func (c *cache) Set(ctx context.Context, policyId string, ruleName string, imageRef string) (bool, error) {
+	c.logger.Info("Setting cache", "policyId", policyId, "ruleName", ruleName, "imageRef", imageRef)
+	if !c.isCacheEnabled {
+		return false, nil
+	}
+	c.logger.Info("Successfully set cache", "policyId", policyId, "ruleName", ruleName, "imageRef", imageRef)
+	return false, nil
+}
+
+func (c *cache) Get(ctx context.Context, policyId string, ruleName string, imageRef string) (bool, error) {
+	c.logger.Info("Searching in cache", "policyId", policyId, "ruleName", ruleName, "imageRef", imageRef)
+	if !c.isCacheEnabled {
+		return false, nil
+	}
+	c.logger.Info("Cache entry not found", "policyId", policyId, "ruleName", ruleName, "imageRef", imageRef)
+	c.logger.Info("Cache entry found", "policyId", policyId, "ruleName", ruleName, "imageRef", imageRef)
```

Comment:
Conflict logs.

Reply:
Oh! I have added that there just as a reminder to add a log for "not found" as well, as the cache has not been implemented yet, should we keep it?